exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -68,5 +79,151 @@ func (prwe *prwExporter) shutdown(context.Context) error {
 // TimeSeries, validates and handles each individual metric, adding the converted TimeSeries to the map, and finally
 // exports the map.
 func (prwe *prwExporter) pushMetrics(ctx context.Context, md pdata.Metrics) (int, error) {
-	return 0, nil
+	prwe.wg.Add(1)
+	defer prwe.wg.Done()
+	select {
+	case <-prwe.closeChan:
+		return pdatautil.MetricCount(md), fmt.Errorf("shutdown has been called")
+	default:
+		tsMap := map[string]*prompb.TimeSeries{}
+		dropped := 0
+		errs := []error{}
+
+		resourceMetrics := data.MetricDataToOtlp(pdatautil.MetricsToInternalMetrics(md))
+		for _, resourceMetric := range resourceMetrics {
+			// TODO: add resource attributes as labels, probably in next PR
+			for _, instrumentationMetrics := range resourceMetric.InstrumentationLibraryMetrics {
+				// TODO: decide if instrumentation library information should be exported as labels
+				for _, metric := range instrumentationMetrics.Metrics {
+					// check for valid type and temporality combination
+					if ok := validateMetrics(metric.MetricDescriptor); !ok {
+						dropped++
+						errs = append(errs, fmt.Errorf("invalid temporality and type combination"))
+						continue
+					}
+					// handle individual metric based on type
+					switch metric.GetMetricDescriptor().GetType() {
+					case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64,
+						otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+						if err := prwe.handleScalarMetric(tsMap, metric); err != nil {
+							dropped++
+							errs = append(errs, err)
+						}
+					}
+				}
+			}
+		}
+
+		if err := prwe.export(ctx, tsMap); err != nil {
+			return pdatautil.MetricCount(md), err
+		}
+
+		if dropped != 0 {
+			return dropped, componenterror.CombineErrors(errs)
+		}
+
+		return 0, nil
+	}
+}
+
+// handleScalarMetric processes data points in a single OTLP scalar metric by adding the each point as a Sample into
+// its corresponding TimeSeries in tsMap.
+// tsMap and metric cannot be nil, and metric must have a non-nil descriptor
+func (prwe *prwExporter) handleScalarMetric(tsMap map[string]*prompb.TimeSeries, metric *otlp.Metric) error {
+
+	mType := metric.MetricDescriptor.Type
+
+	switch mType {
+	// int points
+	case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64:
+		if metric.Int64DataPoints == nil {
+			return fmt.Errorf("nil data point field in metric %v", metric.GetMetricDescriptor().Name)
+		}
+
+		for _, pt := range metric.Int64DataPoints {
+
+			// create parameters for addSample
+			name := getPromMetricName(metric.GetMetricDescriptor(), prwe.namespace)
+			labels := createLabelSet(pt.GetLabels(), nameStr, name)
+			sample := &prompb.Sample{
+				Value: float64(pt.Value),
+				// convert ns to ms
+				Timestamp: convertTimeStamp(pt.TimeUnixNano),
+			}
+
+			addSample(tsMap, sample, labels, metric.GetMetricDescriptor().GetType())
```

Comment:
I did not seem to see that addSample treats "MONOTONIC_" as counters. Am I missing this?

Reply:
Yes, I am treating `MONOTONI_` and non-monotonic the same way and assuming current value is already calculated. Is this what you are referring to? The only other different thing I did for `MONOTONIC_`  is to add a `_total` suffix in its name. https://github.com/open-o11y/opentelemetry-collector/blob/166624cad9f682d2982cea442f5b10e5e1b92ed2/exporter/prometheusremotewriteexporter/helper.go#L178

Reply:
The Prometheus remote write protocol doesn't make this distinction. (I'm not sure how this is handled in a complete system.)

Reply:
https://www.robustperception.io/how-does-a-prometheus-counter-work. Here is a article explaining how Prometheus server handles a counter value reset.